enhancements/machine-api/machine-health-checking.md

```diff
@@ -249,3 +263,175 @@ I want automated remediation on some nodes, but not all -- I may have "pet" mach
 Solution: Create a MachineDisruptionBudget targeting both the "cattle" and "pets" with appropriate thresholds. This is not possible without MachineDisruptionBudget or, possibly, one-off opt-out annotations.
 
 ## Infrastructure Needed
+
+## Related Information
+
+### How does a Machine Health Check determine which Machines it should target for health checking?
+
+A MachineHealthCheck uses a `selector` to specify which Machines it should consider as its targets.
+The selector supports [set-based requirements](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements)
+so there are three options for specifying a selector for a Machine Health Check.
+
+#### An Empty Selector
+
+An empty selector matches all Machines. This means it would include the Control-Plane Machines,
+all Machines created by the default MachineSets and any Machines that users create post installation.
+
+#### Using exact labels
+
+Using the `matchLabels` part of the selector allows specifying a list of key value pairs to check the presence of.
+This means that Machines must have all of the labels specified to be considered to match the selector.
+This could be used to specify that the MHC selects all Machines with the `cluster-api-machine-role: worker` label,
+which would match all MachineSets created by the installer, but not Control-Plane nodes,
+nor necessarily any MachineSets created by users post installation.
+
+#### Using expressions
+
+Using the `matchExpressions` part of the selector allows more complicated expressions to be created for selecting on labels.
+Each expression consists of a key, operator and values.
+This allows users to select a label with value that matches a set of values (operator: In),
+a label with value that is not in a set of values (operator: NotIn). Or check that a label exists or not.
+
+These expressions could be used to select all Machines that do not have the label `cluster-api-machine-role: master`,
+ie, all Machines in the cluster that are not part of the Control-Plane.
+This would include all MachineSets created by the installer and any created by users post installation.
+
+### How does the ClusterAutoscaler interact with MachineHealthCheck?
+
+The ClusterAutoscaler uses its own health checking to determine if a MachineSet is unhealthy after scaling up.
+Should it determine the MachineSet to be unhealthy,
+it would scale the MachineSet back down and try another MachineSet to add new capacity to the cluster.
+
+MachineHealthChecks could interfere with this health checking logic and prevent the autoscaler from adding new compute capacity.
+To ensure there are no conflicts, several scenarios must be tested to check whether there may be conflicts
+
+#### Scenarios
+##### Machine fails to launch due to user config error
+
+Steps to reproduce:
+- Create cluster on AWS
```

Comment:
suggestion, change this to an ordered list just to make it clear these are instructions.